src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -219,11 +219,15 @@
   "Initializes a 'worker' which will listen for lifecycle-related tasks and perform
   them on a background thread, to ensure that we aren't executing multiple lifecycle
   tasks simultaneously."
-  [lifecycle-chan :- (schema/protocol async-prot/Channel)
+  [lifecycle-channel :- (schema/protocol async-prot/Channel)
+   shutdown-channel :- (schema/protocol async-prot/Channel)
    shutdown-reason-promise :- IDeref]
+  (log/debug "Initializing lifecycle worker loop.")
   (async/go (try
               (loop []
```

Comment:
I don't think that's the same.. a go block happens on the core.async thread pool, so I wouldn't expect exception to bubble out.